src/test/java/org/folio/readingroom/controller/PatronPermissionControllerTest.java

```diff
@@ -0,0 +1,90 @@
+import org.folio.spring.service.SystemUserScopedExecutionService;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+
+class PatronPermissionControllerTest extends BaseIT {
+
+  @MockBean
+  UsersClient usersClient;
+  @MockBean
+  private PatonPermissionsRepository patonPermissionsRepository;
+  @Autowired
+  private ReadingRoomRepository readingRoomRepository;
+  @Autowired
+  private SystemUserScopedExecutionService systemUserScopedExecutionService;
+
+  @Test
+  void testSuccessUpdatePatronPermission() throws Exception {
+    ReadingRoom readingRoom = createReadingRoom(READING_ROOM_ID, false);
+    var servicePoints = Set.of(createServicePoint(SERVICE_POINT_ID1, SERVICE_POINT_NAME1),
+      createServicePoint(SERVICE_POINT_ID2, SERVICE_POINT_NAME2));
+    readingRoom.servicePoints(servicePoints);
+
+    this.mockMvc.perform(
+      post("/reading-room")
+          .content(asJsonString(readingRoom))
+          .headers(defaultHeaders())
+          .contentType(MediaType.APPLICATION_JSON)
+          .accept(MediaType.APPLICATION_JSON))
+      .andExpect(status().isCreated());
+
+    PatronPermission patronPermission = createPatronPermission(UUID.randomUUID(),
+      READING_ROOM_ID, UUID.fromString("2205005b-ca51-4a04-87fd-938eefa8f6df"));
+    this.mockMvc.perform(
+      put("/reading-room-patron-permission/2205005b-ca51-4a04-87fd-938eefa8f6df")
+          .content(asJsonString(patronPermission))
+          .headers(defaultHeaders())
+          .contentType(MediaType.APPLICATION_JSON)
+          .accept(MediaType.APPLICATION_JSON))
+      .andExpect(status().is2xxSuccessful());
+  }
+
+  @Test
+  void testUserMismatchIdsUpdatePatronPermission() throws Exception {
+    PatronPermission patronPermission = createPatronPermission(UUID.randomUUID(),
+      UUID.randomUUID(), UUID.fromString("2205005b-ca51-4a04-87fd-938eefa8f6df"));
+    this.mockMvc.perform(
+        put("/reading-room-patron-permission/2205005b-ca51-4a04-87fd-939eefa8f6df")
+          .content(asJsonString(patronPermission))
+          .headers(defaultHeaders())
+          .contentType(MediaType.APPLICATION_JSON)
+          .accept(MediaType.APPLICATION_JSON))
+      .andExpect(status().is4xxClientError());
+  }
+
+  @Test
+  void testUserNotFoundUpdatePatronPermission() throws Exception {
+    PatronPermission patronPermission = createPatronPermission(UUID.randomUUID(),
+      UUID.randomUUID(), UUID.fromString("2305005b-ca51-4a04-87fd-938eefa8f6df"));
+    this.mockMvc.perform(
+        put("/reading-room-patron-permission/2305005b-ca51-4a04-87fd-939eefa8f6df")
+          .content(asJsonString(patronPermission))
+          .headers(defaultHeaders())
+          .contentType(MediaType.APPLICATION_JSON)
+          .accept(MediaType.APPLICATION_JSON))
+      .andExpect(status().is4xxClientError());
+  }
```

Comment:
we can also add one more error scenario
Try to update a patron permission with invalid reading room Id.